### ISSUE TITLE
I used the deconvnet for semantic segmentation, the training looks fine, but at test time, why the output channels are totally black or white?

### ISSUE THREAD

**Issue by brisker** [amrit110/masterThesis#5]
The training looks fine with 10 epoch's iteration
In test time, I have tried to input in the SpatialSoftMax Network with a 2-d tensor or a 3-d tensor, but it always returns the following error:

/home/jcc/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 1 module of nn.Sequential:
/home/jcc/torch/install/share/lua/5.1/nn/SpatialSoftMax.lua:6: attempt to index field 'output' (a nil value)
stack traceback:
	/home/jcc/torch/install/share/lua/5.1/nn/SpatialSoftMax.lua:6: in function </home/jcc/torch/install/share/lua/5.1/nn/SpatialSoftMax.lua:3>
	[C]: in function 'xpcall'
	/home/jcc/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	/home/jcc/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
	[string "require 'cunn'..."]:55: in main chunk
	[C]: in function 'xpcall'
	/home/jcc/torch/install/share/lua/5.1/itorch/main.lua:210: in function </home/jcc/torch/install/share/lua/5.1/itorch/main.lua:174>
	/home/jcc/torch/install/share/lua/5.1/lzmq/poller.lua:75: in function 'poll'
	/home/jcc/torch/install/share/lua/5.1/lzmq/impl/loop.lua:307: in function 'poll'
	/home/jcc/torch/install/share/lua/5.1/lzmq/impl/loop.lua:325: in function 'sleep_ex'
	/home/jcc/torch/install/share/lua/5.1/lzmq/impl/loop.lua:370: in function 'start'
	/home/jcc/torch/install/share/lua/5.1/itorch/main.lua:389: in main chunk
	[C]: in function 'require'
	(command line):1: in main chunk
	[C]: at 0x00406670

Why this happens? I make sure the output of the deconvnet is not nil value
The test code I use is like this: 

![image](https://cloud.githubusercontent.com/assets/13804492/20745758/a52a2d28-b71d-11e6-9a01-e2994c1c7e04.png)
the error just happens in the last line

**Comment by jonla** [comment]
softMaxNet:add(nn.SpatialSoftMax) -> softMaxNet:add(nn.SpatialSoftMax())

**Comment by brisker** [comment]
@jonla 
Thanks

**Comment by brisker** [comment]
@jonla   @amrit110 
After training a model, I find the training epoch seems not enough, so I reload it and try to finetune(retrain), but why the training loss seems like the beginning training loss (from scratch)? 
The code is just different from the training from scratch code here:
model=torch.load() and model=create_model()
Help?
